### PR TITLE
Fix logging

### PR DESCRIPTION
**Pull request by another-rex**
Currently all logs default to the default severity since it's being gathered from stderr. This routes the logs properly by using the google cloud logging library.

### CODE REVIEW

File: gcp/api/server.py [hunk 1]
```diff
@@ -690,9 +691,19 @@ def serve(port: int, local: bool):
     server.stop(_SHUTDOWN_GRACE_DURATION)
 
 
+def is_cloud_run() -> bool:
+  """Check if we are running in cloud run"""
```
Comment (andrewpollock) [inline gcp/api/server.py:695]:
```suggestion
  """Check if we are running in Cloud Run."""
```
Pedantic nit.

**Review by andrewpollock — APPROVED** [review]
Thanks for this. Based on https://cloud.google.com/profiler/docs/about-profiler we should rewrite the API in Go...

File: gcp/api/server.py [hunk 2]
```diff
@@ -690,9 +691,19 @@ def serve(port: int, local: bool):
...
+  # https://cloud.google.com/run/docs/container-contract#env-vars
+  return os.getenv('K_SERVICE') is not None
+
+
 def main():
   """Entrypoint."""
-  logging.basicConfig(stream=sys.stderr)
+  if is_cloud_run():
+    logging_client = google.cloud.logging.Client()
+    logging_client.setup_logging()
```
Comment (michaelkedar) [inline gcp/api/server.py:704]: We could use https://github.com/google/osv.dev/blob/master/osv/logs.py which also surfaces error logs (without stack traces) to the error reporting dashboard.

Comment (another-rex) [reply gcp/api/server.py]: Good point! Imported that function to use instead.

**Review by andrewpollock — APPROVED** [review]

**Review by andrewpollock — APPROVED** [review]
Adjusted PR title to match new reality

File: gcp/api/run_tests.sh [hunk 3]
```diff
@@ -18,12 +18,13 @@ if [ $# -lt 1 ]; then
...
 
-pipenv requirements > requirements.txt
-virtualenv ENV
-. ENV/bin/activate
-pip install -r requirements.txt
+# pipenv requirements > requirements.txt
+# virtualenv ENV
+# . ENV/bin/activate
+# pip install -r requirements.txt
+python3 -m pipenv sync
```
Comment (michaelkedar) [inline gcp/api/run_tests.sh:25]: nit: Want to delete the commented lines?
Any idea why it used virtualenv instead of pipenv in the first place?

**Review by michaelkedar — APPROVED** [review]

File: gcp/api/run_tests.sh (hunk 3 above)
Comment (another-rex) [reply gcp/api/run_tests.sh:25]: Not sure why, I asked Oliver a few weeks ago and he can't remember exactly why either, I'll remove it.

**Pull request merged**